spanner/value.go

```diff
@@ -113,8 +114,26 @@ var (
 	commitTimestamp = time.Unix(0, 0).In(time.FixedZone("CommitTimestamp placeholder", 0xDB))
 
 	jsonNullBytes = []byte("null")
+
+	jsonProvider = jsoniter.ConfigCompatibleWithStandardLibrary
+
+	once sync.Once
 )
 
+// UseNumberWithJSONDecoderEncoder specifies whether Cloud Spanner JSON numbers are decoded
+// as Number (preserving precision) or float64 (risking loss).
+// Defaults to float64, call this method for lossless precision.
+// NOTE: This change affects all clients created by this library, both existing and future ones.
+func UseNumberWithJSONDecoderEncoder() {
```

Comment:
This API means that you can never disable it after having enabled it. I don't think we have a good reason for that restriction, or? I can imagine that making it possible to both enable and disable the option will make it easier for users to use it in their own tests to determine what the behavior is.

So I think that we should add an argument to the method (true/false), or add a separate method for disabling the option.

Reply:
Allowing users to enable/disable the feature means they can run into situation where Client 1(supposed to run without Number) running in thread1 get impacted by some thread2 which uses another Client2 and call UseNumberWithJSONDecoderEncoder(), are we ok with it?

Reply:
That was already the case, however only in the direction `default behavior => new behavior`. Now we allow this to happen both ways.

What I mean is that the original implementation also allowed the following:
1. A client is created. This client is assumed to use the default Go encoding/decoding.
2. The flag is set to true and a new client is created.
3. Both the old client and the new client now use the new behavior.
4. It was however not possible to go back to the default behavior.